tests/core/variant/test_variant.h

```diff
@@ -1787,31 +1790,38 @@ TEST_CASE("[Variant] Writer recursive array") {
 	a2.clear();
 }
 
-TEST_CASE("[Variant] Writer and parser dictionary") {
-	// d = {{1: 2}: 3, 4: "hello", 5: {null: []}}
-	Dictionary d = build_dictionary(build_dictionary(1, 2), 3, 4, String("hello"), 5, build_dictionary(Variant(), build_array()));
-	String d_str;
-	VariantWriter::write_to_string(d, d_str);
+TEST_CASE("[Variant] Writer and parser typed Variant array") {
+	collection_writer_parser(Array(build_array(1, String("hello"), build_array(Variant())), Variant::NIL, "", Variant::NIL), "[1, \"hello\", [null]]");
```

Comment:
The "script" arguments should be passed `Variant()` instead of `Variant::NIL`. While the enum *technically* works in this context, the intent is to take a `Ref<Script>`.
```suggestion
	collection_writer_parser(Array(build_array(1, String("hello"), build_array(Variant())), Variant::NIL, "", Variant()), "[1, \"hello\", [null]]");
```